connector/connect/src/main/protobuf/spark/connect/base.proto

```diff
@@ -48,6 +72,9 @@ message Request {
   // The logical plan to be executed / analyzed.
   Plan plan = 3;
 
+  // (Required) For analyzePlan rpc calls, configure the mode to explain plan in strings.
+  ExplainMode explain_mode= 4;
```

Comment:
should we group the fields in `Request`?

I notice that there are more methods seems should be placed in `AnalyzePlan`:

- isLocal
- isStreaming
- inputFiles
- semanticHash
- catalog APIs (not sure about this)

Reply:
also cc @HyukjinKwon

Reply:
I actually feel like we should start to add more RPC methods. Maybe we should flatten AnalyzePlan to more methods so we don't overload protos....

Reply:
`df.show` can be a proto message, why not `df.explain`?

Reply:
hmmm yeah I guess we can go to the flatten approach which extracts sutff into message.

Reply:
I make `Explain` as a message now. However given how we split  the rpc call, this message needs to stay in `base.proto`:
```
// Main interface for the SparkConnect service.
service SparkConnectService {

  // Executes a request that contains the query and returns a stream of [[Response]].
  rpc ExecutePlan(Request) returns (stream Response) {}

  // Analyzes a query and returns a [[AnalyzeResponse]] containing metadata about the query.
  rpc AnalyzePlan(Request) returns (AnalyzeResponse) {}
}
```

There might be better way to organize it.

Reply:
> df.show can be a proto message, why not df.explain?

I think we can put all functions into `relations.proto`, but one difference is that `df.show` will trigger a job, while `df.explain` only analyzes, maybe better to move it to `AnalyzePlan`